### ISSUE TITLE
Provide repackage for rpm, dep, chocolatey

### ISSUE THREAD

**Issue by jvalkeal** [spring-projects/spring-boot#8267]
As a user it'd be rather nice if you'd get OOB installable package instead of just a fat-jar.

`Spring Cloud Dataflow Server for Yarn` is packaged into rpm [build.gradle](https://github.com/spring-cloud/spring-cloud-dataflow-ambari-rpm/blob/master/build.gradle) using netflix `ospackage` gradle plugin. It'd be very convenient to get these directly out from a boot build. `ospackage` can build `rpm` and `dep` packages, thought there is also a [chocolatey.org](https://chocolatey.org/) for windows and they also have a gradle plugin(though I'm yet to try how this works in a windows world.

I might be able to contribute some cycles to this if you guys think this is something to pursue for?

**Comment by wilkinsona** [comment]
We considered this in the past and decided not to do it: #714. The end goal there was somewhat different though. That said, I still don't think this belongs in Boot. I'd rather people just use the various existing plugins that provide support for packaging In a variety of formats.

**Comment by jvalkeal** [comment]
Yeah, it's indeed being something what I've also thought if it's better to just use existing plugins, and require user to go though that route, vs trying to incorporate all that magic in boot plugin. Just wanted to throw idea out there to get other opinions. Looks like #714 ended with maven not having not so easy integrations with various package tools.

It's just that it'd be so incredibly cool if boot could throw out rpm/dep which you could just scp to target host and tell OS to start it.

**Comment by philwebb** [comment]
I don't think we should do this. It's potentially a lot of work, likely to be difficult to test and isn't really an area where I think Boot should go.
